web/src/components/PasswordWithMeter.js

```diff
@@ -5,7 +5,10 @@ import zxcvbn from '../lazy/zxcvbn';
 import Choice from './Choice';
 
 class Password extends Component {
-  state = { showPassword: false, strength: 0 };
+  constructor() {
```

Comment:
I was able to convert this to a functional component but then the tests absolutely would not pass. There's still an issue with Enzyme's support for the `useEffect` hook, and in this component, `useEffect` would alter the state, triggering a re-render. But Enzyme would bail out before the second render and then tests would fail. Rather than spend all day trying to figure that out, it seemed better to just add a constructor.

Reply:
"Upgrade this, but only if it doesn't also do that, and only during a full moon on a Tuesday."